### PR TITLE
[HACK] Support gzip compression of logs

### PR DESCRIPTION
**Pull request by theunamedguy**
Probably not the "proper" way to do it, but it works well enough. Shrinks my logs by ~70%.

TODO: investigate making compression optional, and modifying any other code that expects logs to be plaintext

### CODE REVIEW

File: modules/Makefile.in [hunk 1]
```diff
@@ -62,7 +62,7 @@ C=
 endif
 
 ifneq "$(LIBZNC)" ""
-LIBS += -L.. -lznc -Wl,-rpath,$(LIBZNCDIR)
+LIBS += -L.. -lznc -lz -Wl,-rpath,$(LIBZNCDIR)
```
Comment (DarthGandalf) [inline modules/Makefile.in]: 1. If you're changing Makefile.in, probably CMakeLists.txt also needs to be updated
2. This line would work only for Cygwin, as other OSes don't use $LIBZNC
3. ZNC already checks whether zlib is available, and turns on its support automatically (for HTTP server)

Just revert this.

File: modules/log.cpp [hunk 2]
```diff
@@ -23,6 +23,7 @@
 #include <znc/Server.h>
 #include <time.h>
 #include <algorithm>
+#include <zlib.h>
```
Comment (DarthGandalf) [inline modules/log.cpp]: We can't make it a required dependency. Surround with `#ifdef HAVE_ZLIB`

File: modules/log.cpp [hunk 3]
```diff
@@ -257,13 +258,19 @@ void CLogMod::PutLog(const CString& sLine,
...
-                      " " + (m_bSanitize ? sLine.StripControls_n() : sLine) +
-                      "\n");
-    } else
+    if (!CFile::Exists(sLogDir))
+        CDir::MakeDir(sLogDir, ModDirInfo.st_mode);
+
+    if (LogFile.Open(O_WRONLY | O_APPEND | O_CREAT))
+    {
+        /* dirty hack */
+        gzFile gzf = gzopen(((std::string)sPath).c_str(), "a");
```
Comment (DarthGandalf) [inline modules/log.cpp]: sPath is already std::string

File: modules/log.cpp [hunk 4]
```diff
@@ -257,13 +258,19 @@ void CLogMod::PutLog(const CString& sLine,
...
-                                         GetUser()->GetTimezone()) +
-                      " " + (m_bSanitize ? sLine.StripControls_n() : sLine) +
-                      "\n");
-    } else
+    if (!CFile::Exists(sLogDir))
+        CDir::MakeDir(sLogDir, ModDirInfo.st_mode);
+
+    if (LogFile.Open(O_WRONLY | O_APPEND | O_CREAT))
+    {
+        /* dirty hack */
```
Comment (DarthGandalf) [inline modules/log.cpp]: What are the options here:
* Keep files open until end of day
* Write uncompressed logs, then compress them at midnight
* Anything else?

That's assuming %d is presented in file name... What to do if it's not?

**Comment by dgw** [comment]
I have a cron script to compress my logs when they reach 30 days old. I'm not convinced this is something that should be built into ZNC itself, especially as a commented "dirty hack". It's simple enough to do from outside ZNC, IMO.

**Comment by jyaworski** [comment]
Ping @theunamedguy. This is pretty useful. Could you look at @DarthGandalf's feedback?

**Comment by theunamedguy** [comment]
@jyaworski: Honestly I've lost interest in this patch myself. Feel free to fork it if you wish.

**Pull request closed without merging**